client/resource_group/controller/limiter.go

```diff
@@ -296,18 +298,23 @@ func (lim *Limiter) Reconfigure(now time.Time,
 ) {
 	lim.mu.Lock()
 	defer lim.mu.Unlock()
-	log.Debug("[resource group controller] before reconfigure", zap.Float64("NewTokens", lim.tokens), zap.Float64("NewRate", float64(lim.limit)), zap.Float64("NotifyThreshold", args.NotifyThreshold))
-	now, _, tokens := lim.advance(now)
-	lim.last = now
-	lim.tokens = tokens + args.NewTokens
+	log.Debug("[resource group controller] before reconfigure", zap.Float64("NewTokens", lim.tokens), zap.Float64("NewRate", float64(lim.limit)), zap.Float64("NotifyThreshold", args.NotifyThreshold), zap.Int64("burst", lim.burst))
+	if args.NewBurst < 0 {
+		lim.tokens = args.NewTokens
+	} else {
+		now, _, tokens := lim.advance(now)
+		lim.last = now
+		lim.tokens = tokens + args.NewTokens
+	}
 	lim.limit = Limit(args.NewRate)
 	lim.burst = args.NewBurst
+	_, _, lim.tokens = lim.advance(now)
```

Comment:
why not advance before `lim.last = now`?

Reply:
I was trying to avoid accumulating too many tokens when burst is -1. Because the first `advance` was before the burst changes, we want to `advance` after the burst changes.
But I feel this worry is unnecessary, I will delete this line